### PR TITLE
feat: expose centroids in approx_percentile_cont fluent api

### LINKED ISSUE #11877: Expose optional `num_centroids` argument in `approx_percentile_cont`'s fluent API
### Is your feature request related to a problem or challenge?

`approx_percentile_cont`'s fluent API [currently exposes](https://docs.rs/datafusion/40.0.0/datafusion/functions_aggregate/approx_percentile_cont/fn.approx_percentile_cont.html) `expression` and `percentile`.

https://github.com/apache/datafusion/blob/1ecdf900786844f09533e91f7e8fd52170ae6ae6/datafusion/functions-aggregate/src/approx_percentile_cont.rs#L49-L55

However, as described in the [datafusion docs](https://datafusion.apache.org/user-guide/sql/aggregate_functions.html#approx-percentile-cont),  the UDAF accepts a 3rd optional argument, an integer number of centroids for T-Digest

https://github.com/apache/datafusion/blob/1ecdf900786844f09533e91f7e8fd52170ae6ae6/datafusion/functions-aggregate/src/approx_percentile_cont.rs#L76-L94

### Describe the solution you'd like

Consistent with exposing optional arguments for [array_slice] and [regexp_*] fluent apis, add a 3rd optional argument to `approx_percentile_cont`'s fluent API.

[array_slice]: https://github.com/apache/datafusion/issues/10424
[regexp_*]: https://github.com/apache/datafusion/pull/10514

### Describe alternatives you've considered

_No response_

### Additional context

I'll have a PR for this shortly.

### PR DESCRIPTION
**Pull request by Michael-J-Ward**
## Which issue does this PR close?

Closes #11877.

## Rationale for this change

The UDAF `approx_percentile_cont` has a [third optional argument](https://datafusion.apache.org/user-guide/sql/aggregate_functions.html#approx-percentile-cont) that the [fluent api](https://docs.rs/datafusion/40.0.0/datafusion/functions_aggregate/approx_percentile_cont/fn.approx_percentile_cont.html) does not currently expose.

This PR add the 3rd optional argument, consistent with how [array_slice](https://github.com/apache/datafusion/issues/10424) and [regexp_*](https://github.com/apache/datafusion/pull/10514) fluent APIs expose optional arguments.

## What changes are included in this PR?

Add `centroids: Option<Expr>` to the fluent API.

## Are these changes tested?

Yes.

## Are there any user-facing changes?

Yes. Any user of the `approx_percentile_cont` fluent API will need to add `None` for the optional argument.

### CODE REVIEW

File: datafusion/functions-aggregate/src/approx_percentile_cont.rs [hunk 1]
```diff
@@ -46,13 +46,21 @@ use datafusion_physical_expr_common::aggregate::tdigest::{
...
-    approx_percentile_cont_udaf
-);
+create_func!(ApproxPercentileCont, approx_percentile_cont_udaf);
+
+/// Computes the approximate percentile continuous of a set of numbers
+pub fn approx_percentile_cont(
+    expression: datafusion_expr::Expr,
+    percentile: datafusion_expr::Expr,
+    centroids: Option<datafusion_expr::Expr>,
+) -> datafusion_expr::Expr {
```
Comment (jayzhan211) [inline datafusion/functions-aggregate/src/approx_percentile_cont.rs:56]: nit: Avoid repeated import prefix

**Review by jayzhan211 — APPROVED** [review]
👍

File: datafusion/core/tests/dataframe/dataframe_functions.rs [hunk 2]
```diff
@@ -394,6 +394,21 @@ async fn test_fn_approx_percentile_cont() -> Result<()> {
 
     assert_batches_eq!(expected, &batches);
 
+    // with number of centroids set
+    let expr = approx_percentile_cont(col("b"), lit(0.5), Some(lit(2)));
```
Comment (Michael-J-Ward) [inline datafusion/core/tests/dataframe/dataframe_functions.rs:398]: NOTE: I don't have SQL Server, so I did not verify that it produces this result.

I chose a choice for `centroids` that produced a different result than the default.

**Comment by alamb** [comment]
I think technically speaking this is an API change as there is a new parameter, so I marked the PR thusly

**Pull request merged**